rust/cubesql/cubesql/src/sql/postgres/shim.rs

```diff
@@ -86,10 +100,16 @@ impl AsyncPostgresShim {
         buffer::write_message(&mut self.socket, message).await
     }
 
-    pub async fn process_startup_message(&mut self) -> Result<bool, Error> {
+    pub async fn process_startup_message(&mut self) -> Result<StartupState, Error> {
         let mut buffer = buffer::read_contents(&mut self.socket).await?;
 
         let startup_message = protocol::StartupMessage::from(&mut buffer).await?;
+
+        if startup_message.protocol_version.major == SSL_REQUEST_PROTOCOL {
```

Comment:
Strictly speaking it makes sense to check both major and minor protocol versions but this will do for now.
Maybe it's a good idea to add a TODO here.